test/jdk/javax/swing/JFileChooser/FileChooserListenerLeak.java

```diff
@@ -52,19 +57,33 @@ public static void main(final String[] args) throws Exception {
         });
     }
 
-    private static void checkListenersCount(JFileChooser chooser) {
-        test(chooser.getComponentListeners());
-        test(chooser.getFocusListeners());
-        test(chooser.getHierarchyListeners());
-        test(chooser.getHierarchyBoundsListeners());
-        test(chooser.getKeyListeners());
-        test(chooser.getMouseListeners());
-        test(chooser.getMouseMotionListeners());
-        test(chooser.getMouseWheelListeners());
-        test(chooser.getInputMethodListeners());
-        test(chooser.getPropertyChangeListeners());
-        test(chooser.getAncestorListeners());
-        test(chooser.getVetoableChangeListeners());
+    private static void checkListenersCount(Component comp) {
```

Comment:
Just a test improvement, now we will check the listeners recursively for each component inside FileChooser.